src/main/java/org/joshsim/lang/interpret/visitor/JoshConfigParserVisitor.java

```diff
@@ -0,0 +1,138 @@
+ */
+public class JoshConfigParserVisitor extends JoshConfigBaseVisitor<JshcFragment> {
+
+  private final EngineValueFactory valueFactory;
+
+  /**
+   * Creates a new visitor with the specified value factory.
+   *
+   * @param valueFactory The factory to use for creating EngineValues
+   */
+  public JoshConfigParserVisitor(EngineValueFactory valueFactory) {
+    this.valueFactory = valueFactory;
+  }
+
+  /**
+   * Visits the root config context and processes all configuration lines.
+   *
+   * @param ctx The ANTLR config context
+   * @return A JshcFragment containing the complete configuration
+   */
+  @Override
+  public JshcFragment visitConfig(JoshConfigParser.ConfigContext ctx) {
+    ConfigBuilder builder = new ConfigBuilder();
+
+    // Process each config line
+    for (JoshConfigParser.ConfigLineContext lineContext : ctx.configLine()) {
+      if (lineContext.assignment() != null) {
+        // Process assignment and add to main builder
+        String variableName = lineContext.assignment().ID().getText();
+        JshcFragment valueFragment = visitValue(lineContext.assignment().value());
+        builder.addValue(variableName, valueFragment.getEngineValue());
+      }
+      // Comments and empty lines are automatically ignored by the grammar
+    }
+
+    return new JshcConfigBuilderFragment(builder);
+  }
+
+  /**
+   * Visits an assignment context and creates a config fragment.
+   *
+   * @param ctx The ANTLR assignment context
+   * @return A JshcFragment containing the assignment
+   */
+  @Override
+  public JshcFragment visitAssignment(JoshConfigParser.AssignmentContext ctx) {
+    String variableName = ctx.ID().getText();
+    JshcFragment valueFragment = visit(ctx.value());
+
+    // Add the value to a new config builder
+    ConfigBuilder builder = new ConfigBuilder();
+    builder.addValue(variableName, valueFragment.getEngineValue());
+
+    return new JshcConfigBuilderFragment(builder);
+  }
+
+  /**
+   * Visits a value context and parses the number and units.
+   *
+   * @param ctx The ANTLR value context
+   * @return A JshcFragment containing the parsed value
+   */
+  @Override
+  public JshcFragment visitValue(JoshConfigParser.ValueContext ctx) {
+    String numberText = ctx.NUMBER().getText();
+    String unitsText = "";
+
+    if (ctx.ID() != null) {
+      unitsText = ctx.ID().getText();
+    }
+
+    try {
+      // Parse the number
+      BigDecimal number = new BigDecimal(numberText);
+
+      // Create units object
+      Units units = unitsText.isEmpty() ? Units.EMPTY : Units.of(unitsText);
+
+      // Create EngineValue
+      EngineValue engineValue = valueFactory.parseNumber(numberText, units);
+
+      return new JshcValueFragment(number, unitsText, engineValue);
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Invalid number format: " + numberText, e);
+    } catch (Exception e) {
```

Comment:
Don't love the catch of all Exceptions. Is there another way to do this or is this needed? If so, why is it needed?